# ???
ControllerType: 1
# ???
w_control_mode: 0
# ???
LineFitting_method: 1


# The mode which is used for tuning the mask (no velocity will be published)
mask_tune: false
# Scale of the image processed (it resizes the input images before process)
Scale: 1.0
# min radius of contour which will be detected and contribute on detecting line 
minContourSize: 2.0
# In case of having only one camera set this param to true (the front camera will be used)
single_camera_mode: false
# controlling the  velocity publisher stare (ON: True of OFF: false)
publish_cmd_vel: true
# controls publishing linear velocity (used fro debug and setting angular PID params)
publish_linear_vel: true
# ???
max_row_num: 20
# ???
div_soft: 25
# -------------------------------------------------------
#---------------------- Neighbourhood---------------------
# Size of input image should be set here 
width: 640
height: 480
# default position of tracking windows 
ex_Xc: 320
ex_Yc: 240
# default size of tracking windows
nh_L: 120
nh_H: 250
# ???
nh_offset: 200
# min point where sate controller will switch the mode inside the tracking windows
min_points_switch: 10
# min point where sate controller will switch the mode inside image
min_frame: 30
# ???
coef: 55
#-------------------------------------------------------
#--------------------- Velocity Topic ------------------ 
# desired forward linear velocity
vf_des: 0.2   
# desired backward linear velocity
vb_des: 0.2   
# Max angular velocity
w_max: 0.1 
# Min angular velocity 
w_min: 0.01
# ???
z_min: 0.01
# -------------------------------------------------------
#---------------------- Controller ---------------------- 
del_t: 0.1
ty: 0.6

tz: 0.7 # 0.7 husky
rho_b: -60.0
rho_f: -60.0
# -------------------------------------------------------
#---------------------- Gains --------------------------- 
# husky
lambda_x_1: 10
lambda_w_1: 1

lambda_x_2: 0
lambda_w_2: 5000


lambda_x_3: 10
lambda_w_3: 1

lambda_x_4: 0
lambda_w_4: 5000
# -------------------------------------------------------

#---------------------- States to start ----------------- 
# uncomment each of these states and controller will start from that state to drive robot
mode: 1
cam_num: 1
drive_forward: true
turning_mode: false
steering_dir: 1
driving_dir: 1

# mode: 2
# cam_num: 2
# drive_forward: false
# turning_mode: true
# steering_dir: -1
# driving_dir: 1

# mode: 3
# cam_num: 2
# drive_forward: true
# turning_mode: false
# steering_dir: -1
# driving_dir: -1

# mode: 4
# cam_num: 1
# drive_forward: false
# turning_mode: true
# steering_dir: 1
# driving_dir: -1
#-------------------------------------------------------
#----------------Image filter params-------------------- 
# field
max_Hue: 80
min_Hue: 40

max_Saturation: 255
min_Saturation: 100

max_Value: 255
min_value: 70

